reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/TimedSchedulerTest.java

```diff
@@ -324,4 +329,65 @@ void workerSchedulePeriodicallyIsCorrectlyMetered() throws InterruptedException
 				+ testScheduler.submittedPeriodicInitial.count()
 				+ testScheduler.submittedPeriodicIteration.count(), "completed tasks == sum of all timer counts");
 	}
+
+	@Test
+	void pendingScheduleRemovedOnScheduleRejection() {
+		CountDownLatch cdl = new CountDownLatch(1);
+		ExecutorService executorService = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
+				new SynchronousQueue<>());
+		Scheduler original = Schedulers.fromExecutorService(executorService);
+		TimedScheduler testScheduler = new TimedScheduler(original, registry, "test", Tags.empty());
+		RequiredSearch requiredSearch = registry.get("test.scheduler.tasks.pending");
+		LongTaskTimer longTaskTimer = requiredSearch.longTaskTimer();
+
+		Runnable supp = () -> {
+			try {
+				cdl.await();
+			} catch (InterruptedException e) {
+				throw new RuntimeException(e);
+			}
+		};
+
+		assertThatNoException().isThrownBy(() -> testScheduler.schedule(supp));
+		assertThatExceptionOfType(RejectedExecutionException.class).isThrownBy(() -> testScheduler.schedule(supp));
+		assertThatExceptionOfType(RejectedExecutionException.class)
+				.isThrownBy(() -> testScheduler.schedule(supp, 0, TimeUnit.SECONDS));
+
+		cdl.countDown();
+
+		assertThat(longTaskTimer.activeTasks())
+				.as("longTaskTimer.activeTasks()")
```

Comment:
It seems you validate it gets zeroed, however the precondition should be to first validate a condition under which the timer is non-zero.